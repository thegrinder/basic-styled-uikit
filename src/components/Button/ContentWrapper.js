import { bool } from 'prop-types';
import styled from 'styled-components';

const propTypes = {
  submitting: bool.isRequired,
};

const ContentWrapper = styled.span`
  display: flex;
  align-items: center;
  opacity: ${({ submitting }) => (submitting ? '0' : '1')};
`;

ContentWrapper.propTypes = propTypes;

export default ContentWrapper;
