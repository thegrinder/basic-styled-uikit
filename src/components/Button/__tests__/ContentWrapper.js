import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import ContentWrapper from '../ContentWrapper';

describe('<ContentWrapper />', () => {
  afterEach(cleanup);

  it('should render correctly', () => {
    const { container: { firstChild } } = render(
      <ContentWrapper submitting={true} />,
    );
    expect(firstChild).toBeDefined();
    expect(firstChild).toMatchSnapshot();
  });

  it('should render <span> tag', () => {
    const { container: { firstChild } } = render(
      <ContentWrapper submitting={true} />,
    );
    expect(firstChild.tagName).toEqual('SPAN');
  });

  it('should render content correctly', () => {
    const content = 'content';
    const { container: { firstChild } } = render(
      <ContentWrapper submitting={true}>
        {content}
      </ContentWrapper>,
    );
    expect(firstChild).toHaveTextContent(content);
    expect(firstChild).toMatchSnapshot();
  });
});
