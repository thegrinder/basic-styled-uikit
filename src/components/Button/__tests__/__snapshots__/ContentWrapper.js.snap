// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentWrapper /> should render content correctly 1`] = `
<span
  class="ContentWrapper-juvto2-0 haOASu"
>
  content
</span>
`;

exports[`<ContentWrapper /> should render correctly 1`] = `
<span
  class="ContentWrapper-juvto2-0 haOASu"
/>
`;
